src/features/SpellCheckerProvider.ts

```diff
@@ -203,6 +203,20 @@ export default class SpellCheckerProvider implements vscode.CodeActionProvider
 			console.log( text );
 			console.log( '------------------------------------------' );
 		}
+		// remove LaTeX commands
+		text = text.replace( /\\\w*{.*?}/g, '' );
+		if( DEBUG )
+		{
+			console.log( text );
+			console.log( '------------------------------------------' );
+		}
+		// remove LaTeX non-breaking spaces
+		text = text.replace( /\\~/g, '' );
```

Comment:
This matches the string "\~" but not "~". The tilde isn't a special character in regexs, so you don't need to escape it. Also, instead of making a new regex, you can add the tilde character to the list of special characters in line 172